### DISCUSSION TITLE
Does APISIX support more advanced protocal like CoAP?

### DISCUSSION THREAD

**Discussion by ITCodd** [apache/apisix#6166] [Ideas]
apisix 能对协议进行扩展吗，比如物联网的协议coap

**Comment by tokers** [comment]
cc @shuaijinchao

**Reply by moonming** [reply]
ping @shuaijinchao

**Comment by shuaijinchao** [comment]
we are currently planning to propose a protocol extension framework xRPC which is based on 4-layer Stream, and we can easily extend any application layer protocol on top of it.

**Comment by moonming** [comment]
Apache APISIX already supports [MQTT](https://github.com/apache/apisix/blob/master/docs/en/latest/plugins/mqtt-proxy.md), and adding support for CoAP is similar, PR is welcome
